libMicro -- Small scrip fixes

andipla opened the pull request:
Hi,

I had 2 small problems building and running libMicro on Debian Jessie, so I fixed them.

I did however not test it on any other distro / OS than Debian Jessie.

regards,
Andi

andipla committed:
fix bench script -- arch does not support -k in Debian

Since the -k parameter is not supported by arch on Debian (tested on Jessie
and Squeeze), the pathname was not correct. Thus the -k is dropped as it
returns (as seems to be expected) the architecture of the compiling machine.

Signed-off-by: Andreas Platschek <andreas.platschek@opentech.at>

portante commented:
Thanks @andipla.

It would be great if we could conditionalize this changes for just Debian for now so that we don't break the other platforms.  Do you think you'd be able to do that?

andipla committed:
Make the arch fix for Debian conditional

This is to insure that other platforms don't break.

Signed-off-by: Andreas Platschek <andreas.platschek@opentech.at>
Signed-off-by: Andreas Gabriel-Platschek <andi.platschek@gmail.com>

portante commented on Makefile.com:
```diff
@@ -107,7 +107,7 @@ tattle:		../tattle.c	libmicro.a
 	echo "char * CC = \""$(CC)"\";" >> tattle.h
 	echo "char * extra_compiler_flags = \""$(extra_CFLAGS)"\";" >> tattle.h
-	$(CC) -o tattle $(CFLAGS) -I. ../tattle.c libmicro.a -lrt -lm
+	$(CC) -o tattle $(CFLAGS) -I. ../tattle.c libmicro.a -lrt -lm -lpthread
```
What is the missing symbol satisfied by adding `-lpthread` here?

andipla commented:
That's pthread_create:

/usr/bin/ld: libmicro.a(libmicro.o): undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
//lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
../Makefile.com:107: recipe for target 'tattle' failed

The orignal version works in Debian wheezy, but not in jessie and stretch, those need the -lpthread

thx!

andipla committed:
Extended the conditional in bench scripts to Ubuntu

While testing this on Ubuntu 14.04.4 LTS I noticed that in Ubuntu
arch does not support the -k option either.

Signed-off-by: Andreas Platschek <andi.platschek@gmail.com>

andipla commented:
I added a condition for the ARCH variable in the bench scripts.

The only system other than Debian wheezy/jessie/stretch I have around to test this is an Ubuntu 14.04 which has the same problems, so I added Ubuntu to the conditional as well.

btw. Ubuntu also had the same problem without -lpthread in the Makefile.com, except it did not only complain about pthread_create but also about pthread_join.

portante commented:
@andipla, I am behind on merging this, bare with me.

portante commented on bench:
```diff
@@ -65,7 +65,14 @@ mkdir -p $VDIR1 $VDIR2
 touch $IFILE
 
-ARCH=`arch -k`
+DISTRO=$(cat /etc/issue | head -n +1 | awk '{print $1}')
```
Can you change this to, `cat /etc/issue 2> /dev/null`, so that on distributions where `/etc/issue` does not exist, the error message won't be emitted?

andipla replied:
sure thing, just pushed it.

andipla committed:
non-Debian users shall not be bothered -> dump error message to /dev/null

Signed-off-by: Andreas Platschek <andreas.platschek@opentech.at>

andipla committed:
cascade_flock: fix implicit declaration of flock()

Signed-off-by: Andreas Gabriel-Platschek <andi.platschek@gmail.com>

andipla committed:
stcmp(): fix implicit declaration in several places

Signed-off-by: Andreas Gabriel-Platschek <andi.platschek@gmail.com>

portante reviewed (commented):
Latest changes look good.  Still wondering about the addition of `-lpthread` and what problem it is solving.

andipla commented:
> Latest changes look good. Still wondering about the addition of `-lpthread` and what problem it is solving.

Sorry, I was unaware I was pushing to an open pull request . Anyhow -- this was a long time ago. I tried to revert that patch and reproduce the issue, but since I am no longer able to reproduce the issue (debian trixie) I seemed to have back then I kicked that one commit out.

portante reviewed (approved):
Looks good.
